core/src/main/scala/org/apache/spark/api/python/StreamingPythonRunner.scala

```diff
@@ -91,11 +91,27 @@ private[spark] class StreamingPythonRunner(
       new BufferedInputStream(pythonWorker.get.channel.socket().getInputStream, bufferSize))
 
     val resFromPython = dataIn.readInt()
+    if (resFromPython != 0) {
+      val errMessage = PythonWorkerUtils.readUTF(dataIn)
+      throw streamingPythonRunnerDidNotInitialize(resFromPython, errMessage)
+    }
     logInfo(s"Runner initialization succeeded (returned $resFromPython).")
 
     (dataOut, dataIn)
   }
 
+  def streamingPythonRunnerDidNotInitialize(resFromPython: Int, errMessage: String):
```

Comment:
minor: rename to '...RunnerInitializationFailure' or '...FailedToInitialize' ...

Reply:
Done